### PR TITLE
feat(cost-model): implement cost model derive statistics

### PR DESCRIPTION
**Pull request by xx01cyx**

### CODE REVIEW

File: optd-cost-model/src/cost_model.rs [hunk 1]
```diff
@@ -40,28 +41,83 @@ impl<S: CostModelStorageManager> CostModelImpl<S> {
...
         predicates: &[ArcPredicateNode],
-        children_stats: &[Option<&EstimatedStatistic>],
+        children_stats: &[EstimatedStatistic],
         context: ComputeCostContext,
     ) -> CostModelResult<Cost> {
         todo!()
     }
 
-    fn derive_statistics(
+    async fn derive_statistics(
```
Comment (unw9527) [inline optd-cost-model/src/cost_model.rs:56]: Make this pub?

File: optd-cost-model/src/cost_model.rs [hunk 2]
```diff
@@ -40,28 +41,83 @@ impl<S: CostModelStorageManager> CostModelImpl<S> {
...
     ) -> CostModelResult<EstimatedStatistic> {
-        todo!()
+        match node {
+            PhysicalNodeType::PhysicalScan => {
+                let table_id = TableId(predicates[0].data.as_ref().unwrap().as_u64());
+                let row_cnt = self
+                    .storage_manager
+                    .get_table_row_count(table_id)
+                    .await?
+                    .unwrap_or(1) as f64;
```
Comment (unw9527) [inline optd-cost-model/src/cost_model.rs:70]: Is the table guaranteed to have at least one row?

Comment (xx01cyx) [reply optd-cost-model/src/cost_model.rs:70]: No. I would say this is sth like a default value. As long as you initialize the storage part with corresponding base table statistics, the code won't hit this line.

File: optd-cost-model/src/cost_model.rs (hunk 1 above)
Comment (xx01cyx) [reply optd-cost-model/src/cost_model.rs:56]: The methods are public because `trait CostModel` is public.

Comment (unw9527) [reply optd-cost-model/src/cost_model.rs:56]: Oh you're right

**Review by lanlou1554 — APPROVED** [review]

**Pull request merged**